File: originMatlabCode/connect_coverage/LaplaMat.m
function [L,A,d]=LaplaMat(x,R,deta)
%输入：x为N*n的状态矩阵，N为代理数量，n为状态维度；R为通信范围
%输出：n*n的拉普拉斯矩阵L与邻接矩阵A
value=-(R^2/log(deta));%2*sigma^2
d=squareform(pdist(x,'euclidean'));%距离矩阵
N=size(x,1);
A=zeros(N,N);
for i=1:N
    for j=1:i
        if i==j
            continue
        end
        if d(i,j)<=R
            A(i,j)=exp(-d(i,j)^2/(value));
        else
            A(i,j)=0;
        end
        A(j,i)=A(i,j); 
    end
end
A=A-diag(diag(A));%邻接矩阵
D=diag(sum(A,2));%度矩阵
L=D-A;%拉普拉斯矩阵
end

File: algorithms/area_coverage/LaplaMat.py
# -*- coding: UTF-8 -*-
#!/usr/bin/env python
import numpy as np
import math
from scipy.spatial.distance import pdist,squareform

#定义计算拉普拉斯矩阵的函数
#输入：x为N*n的状态矩阵，N为智能体数量，n为状态维度；R为通信范围
#输出：n*n的拉普拉斯矩阵L，邻接矩阵A，距离矩阵d
def L_Mat(X,R,deta):
   value=-(R**2/math.log(deta))
   d = squareform(pdist(X,'euclidean'))    
   N = len(X)
   A = np.zeros((N,N))
   for i in range(N):
       for j in range(N):
           if(i==j):
               continue
           if(d[i,j]<=R):
               A[i,j] = math.exp(-d[i,j]**2/(value))
           else:
               A[i,j] = 0
           A[i,j] = A[i,j] 
   A = A-np.diag(np.diag(A))
   D = np.diag(np.sum(A,axis=1))
   L = D-A
   
   return L,A,d
   


